### DISCUSSION TITLE
RTSP via UDP results in 461 Unsupported Transport

### DISCUSSION THREAD

**Discussion by stwalser** [bluenviron/mediamtx#2871] [Questions]
### Question

I have a network TV tuner in my home network that provides RTSP streams of the different channels via UDP on different URLs.
When I try to access one of the streams via mediamtx, I get the error 461 Unsupported Transport from my network TV tuner. However, when I access the stream via ffprobe,  no error occurs. 

I set `rtspTransport: udp` in the `.yml` file of the server to enforce UDP as the transport protocol, because the tuner does not support TCP.
My server version is 1.4.1.

This is the debug log:

> 2024/01/04 11:00:17 DEB [path mystream] [RTSP source] [c->s] OPTIONS rtsp://10.0.0.19:554/?src=1&freq=11303&pol=h&msys=dvbs2&sr=22000&x_ci=1&x_pmt=107.4911&pids=0,16,17,18,20,107,1920,1921,1922,1925 RTSP/1.0
> CSeq: 1
> User-Agent: gortsplib
> 
> 
> 2024/01/04 11:00:17 DEB [path mystream] [RTSP source] [s->c] RTSP/1.0 200 OK
> CSeq: 1
> Public: OPTIONS, DESCRIBE, SETUP, PLAY, TEARDOWN
> 
> 
> 2024/01/04 11:00:17 DEB [path mystream] [RTSP source] [c->s] DESCRIBE rtsp://10.0.0.19:554/?src=1&freq=11303&pol=h&msys=dvbs2&sr=22000&x_ci=1&x_pmt=107.4911&pids=0,16,17,18,20,107,1920,1921,1922,1925 RTSP/1.0
> Accept: application/sdp
> CSeq: 2
> User-Agent: gortsplib
> 
> 
> 2024/01/04 11:00:17 DEB [path mystream] [RTSP source] [s->c] RTSP/1.0 200 OK
> CSeq: 2
> Content-Length: 239
> Content-Type: application/sdp
> 
> v=0
> o=- 9876543210 1 IN IP4 10.0.0.19
> s=MPEG TS
> t=0 0
> m=video 0 RTP/AVP 33
> c=IN IP4 0.0.0.0
> a=control:rtsp://10.0.0.19:554/?src=1&freq=11303&pol=h&msys=dvbs2&sr=22000&x_ci=1&x_pmt=107.4911&pids=0,16,17,18,20,107,1920,1921,1922,1925
> 
> 2024/01/04 11:00:17 DEB [path mystream] [RTSP source] [c->s] SETUP rtsp://10.0.0.19:554/?src=1&freq=11303&pol=h&msys=dvbs2&sr=22000&x_ci=1&x_pmt=107.4911&pids=0,16,17,18,20,107,1920,1921,1922,1925 RTSP/1.0
> CSeq: 3
> Transport: RTP/AVP;unicast;client_port=58280-58281;mode=play
> User-Agent: gortsplib
> 
> 
> 2024/01/04 11:00:17 DEB [path mystream] [RTSP source] [s->c] RTSP/1.0 461 Unsupported Transport
> CSeq: 3
> 
> 
> 2024/01/04 11:00:17 ERR [path mystream] [RTSP source] bad status code: 461 (Unsupported Transport)

Is there a different configuration I am missing to get this to work or is this a setup which mediamtx does not support?

EDIT:
Some more debugging yielded a difference in the transport field of the SETUP header call:
ffprobe: Transport: RTP/AVP/UDP;unicast;client_port=9368-9369
mediamtx: Transport: RTP/AVP;unicast;client_port=18348-18349;mode=play

Is it possible to modify the header of mediamtx such that it equals the ffprobe header?

**Comment by aler9** [comment]
Hello, in order to get this fixed, open a bug report and provide a network dump of communications between FFmpeg (not ffprobe) and the TV tuner by following instructions.
